### PR TITLE
Improve error message

### PR DESCRIPTION
**Pull request by prashantgolash**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug

> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
If both "name" and "generateName" are present in objectMeta YAML spec, and an object with the same "name" already exists, the error thrown during creation is that of ServerTimeout. Essentially, "generateName" is ignored if "name" exists in the spec, so ideally the error shouldn't be related to "generateName" (i.e. timeout). This PR intends to solve that by setting "generateName" to empty if the "name" is already present.

**Which issue(s) this PR fixes**:

Fixes # 32220

**Special notes for your reviewer**:
This PR is carry forward work of the previous PR discussed here - https://github.com/kubernetes/kubernetes/pull/61642. I put some of my comments in this note - https://github.com/kubernetes/kubernetes/issues/32220.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by prashantgolash** [comment]
I signed it

File: staging/src/k8s.io/apiserver/pkg/registry/rest/create_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	for _, test := range tests {
+		BeforeCreate(mock, ctx, test.resource)
+		objectMeta, _, kerr := objectMetaAndKind(mock, test.resource)
+		if kerr != nil {
+			t.Fatal(kerr)
+		}
+
+		if len(objectMeta.GetName()) > 0 {
+			if objectMeta.GetGenerateName() != "" {
+				t.Fatal("Generate name has to empty, if the resource has name")
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/create_test.go]: `Generate name` -> `GenerateName`

File: staging/src/k8s.io/apiserver/pkg/registry/rest/create_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+			}
+		}
+
+		if test.resource.GetGenerateName() != objectMeta.GetGenerateName() {
+			t.Fatalf("error in generate name expected %s got %s", test.resource.GetGenerateName(), objectMeta.GetGenerateName())
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/create_test.go]: `"error in GenerateName; expected %s, got %s"`

**Review by neolit123 — COMMENTED** [review]
thanks for the PR, it seems that your CLA is not working. please try contacting the CNCF help desk.
also squash the commits.

**Comment by prashantgolash** [comment]
@neolit123 - sorry for the mess up. My user name got messed up which was preventing cnf license. I will reopen a new pull request. Apologize for the noise.

**Pull request closed without merging**